src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -468,30 +482,89 @@ public void postUsersOutboxProcess(Map<String, String> okapiHeaders, Handler<Asy
       });
   }
 
-  private void updateUser(User entity, Map<String, String> okapiHeaders,
+  private void updateUser(User entity, Map<String, String> okapiHeaders, PostgresClient pgClient,
       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
     Date now = new Date();
     entity.setCreatedDate(now);
     entity.setUpdatedDate(now);
-    PgUtil.put(TABLE_NAME_USERS, entity, entity.getId(), okapiHeaders, vertxContext, PutUsersByUserIdResponse.class, reply -> {
-      if (isDuplicateUsernameError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "User with this username already exists")));
-        return;
-      }
-      if (isDuplicateBarcodeError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "This barcode has already been taken")));
-        return;
-      }
-      logger.debug("Save successful");
-      asyncResultHandler.handle(reply);
-    });
+
+    pgClient.withTrans(conn -> usersService.getUserById(conn, entity.getId())
+        .compose(userFromStorage -> {
+          if (userFromStorage == null) {
+            return succeededFuture(PutUsersByUserIdResponse.respond404WithTextPlain(entity.getId()));
+          }
+          boolean isConsortiaFieldsUpdated = isConsortiumUserFieldsUpdated(entity, userFromStorage);
+
+          return usersService.updateUser(conn, entity)
+            .compose(user -> {
+              if (isConsortiaFieldsUpdated) {
+                return userOutboxService.saveUserOutboxLog(conn, user, UserEvent.Action.EDIT, okapiHeaders);
+              }
+              return Future.succeededFuture();
+            })
+            .map(aVoid -> PutUsersByUserIdResponse.respond204())
+            .map(Response.class::cast);
+        })
+      .onComplete(reply -> {
+
+        if (reply.cause() != null) {
+          handleUpdateUserFailures(asyncResultHandler, reply);
+          return;
+        }
+
+        userOutboxService.processOutboxEventLogs(vertxContext.owner(), okapiHeaders);
+        asyncResultHandler.handle(reply);
+      }));
+  }
+
+  private void handleUpdateUserFailures(Handler<AsyncResult<Response>> asyncResultHandler, AsyncResult<Response> reply) {
+    String errorMessage = reply.cause().getMessage();
+    if (isDuplicateUsernameError(errorMessage)) {
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "User with this username already exists")));
+      return;
+    }
+
+    if (isDuplicateBarcodeError(errorMessage)) {
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "This barcode has already been taken")));
+      return;
```

Comment:
An error log may help over here ?

Reply:
Added log.

Reply:
If we use logging here and in previous statement, I propose to use info level, because actually these cases are not real errors. So for example when we have some issues basically support team will send WARN and ERROR level logs to us, and error logs like "username is exists" can confuse with real errors